packages/reporters/reporters/migrations-V5/reporter.js

```diff
@@ -367,35 +369,21 @@ class Reporter {
   async postDeploy(data) {
     let message;
     if (data.deployed) {
-      const tx = await data.contract.interfaceAdapter.getTransaction(
-        data.receipt.transactionHash
-      );
-
-      const block = await data.contract.interfaceAdapter.getBlock(
-        data.receipt.blockNumber
-      );
-
-      // if geth returns null, try again!
-      if (!block) return this.postDeploy(data);
-
-      data.timestamp = block.timestamp;
-
-      const balance = await data.contract.interfaceAdapter.getBalance(tx.from);
+      const { convertCost, ...txData } = await data.contract.interfaceAdapter.getTransactionReportData(data.receipt);
```

Comment:
This makes me think that we should just move `interfaceAdapter` to the reporter itself. Maybe that can be separate.

Reply:
Not sure I'm following you here. Do you mean that reporter should have its own `interfaceAdapter` when it's instantiated instead of using the one from the contract?

It would limit the ability to use different `interfaceAdapter` per contract, but that's something that wouldn't work anyway as they need to have the same cost unit to add them up.

Reply:
It would also sort the issue I mentioned on the previous comment, about `postMigrate` not having access to `interfaceAdapter` right now.

Reply:
Oh I see. I did not consider that different contracts can have different interface adapters.

Reply:
Thanks for the insight!